### DISCUSSION TITLE
Bug with -Canny cropping unintentionally ?

### DISCUSSION THREAD

**Discussion by marshallapparalautomation** [ImageMagick/ImageMagick#6174] [Help] [answered]
### ImageMagick version

7.1.1-3

### Operating system, version and so on

Mac Ventura

### Description

I'm playing with -canny and the resulting image is being scaled up or cropped unintentionally. Seems like a bug. Starting and Ending image are the same size and resolution, however the result seems about 5% larger. Is there a way to prevent this?

**Comment by fmw42** [comment]
Please provide your input and output image and your exact command line.  When you say larger, do you mean dimensions or file size?

**Reply by marshallapparalautomation** [reply]
I mean the image is larger visually inside the boundary of the original image (CROPPING). I think it's related somehow to transparency in PNGs. Doesn't happen to a png without transparence. See the 2 before an afters...see how the image with transparency no longer matches the cropping/size of the infile.

Command: magick infile.png   -colorspace gray -canny 0x2+.02%+10%  outfile.png

![OutTransparentTest](https://user-images.githubusercontent.com/68432172/226135294-043fcd2c-68f8-4e73-8468-040d96acb67f.png)
![InfileTransparent](https://user-images.githubusercontent.com/68432172/226135295-3ab2fa99-68db-4cae-9d73-d65d539712a3.png)
![outfileflat](https://user-images.githubusercontent.com/68432172/226135299-8193d99c-09a3-46d5-9b88-fa652ba86f95.png)
![infileFlat](https://user-images.githubusercontent.com/68432172/226135300-5e71f1e0-0825-445c-962c-8682959e926c.jpg)

**Comment by fmw42** [comment]
You could floodfill your canny image with some color other than white and black starting at the center.  Then convert everything but that color to black and that color to white. Then trim that image to get the crop values and use them for the input.

When you crop to a PNG output use +repage after the crop to remove the virtual canvas that PNG keeps about the image.

**Reply by marshallapparalautomation** [reply]
Does trim return the crop values and position as a result? I think in this case I would not want to use +repage since that's how Ill be able to locate x,y position of the inside shape correct?

**Comment by fmw42** [comment, marked as answer]
What you want is the virtual trim command I showed above using %@. That returns WxH+X+Y.

When I said use +repage, that was for when you complained your PNG result was too large for the crop. After -crop use +repage on PNG and TIFF and GIF which keep a virtual canvas .

**Reply by marshallapparalautomation** [reply]
Thank you!
